documentation/design/proposals/management_tasks-for-shared.adoc

```diff
@@ -0,0 +1,88 @@
+= Management Tasks for Shared
+
+EnMasse user stories defines a number of management functions that are available to the Tenant Admin.  These include
+purging a queue/subscription, closing a connection,  move/copy message(s) from one queue/subscription to a different address.
+
+Of those use-cases, the current solution implements only purge. The current implementation uses synchronous request/responses
+messages (HTTP and AMQP) to action the managemnet requests and relies on the existance of the `agent` component (something
+that will disappear with shared infrastructure.
+
+We need new design for this part of the system in order to deliver shared.
+
+== Current Design Flaws
+
+The current design has number of drawbacks:
+
+1. The design assumes that the management request can be completed within a HTTP request timeout (of both the use-agent
+   and HTTP proxies).  If the management request takes too long the response channel is lost meaning the user has no
+   mechanism to monitor the request's completion.  Purging a deep queue or moving a significant number of messages
+   is likely to breach timeouts. 
+1. The deisgn has no mechnaism to monitor the progress of a request
+1. The deisgn has no mechnaism to cancel a management request.
+1. The Console UI uses modals UI elements to initiate the request and blocks awaiting the response. The user is unable
+   to continue to interact with the app whilst the request is in progress.
+1. There is no provision for a command line interface.
+1. Poorly defined RBAC model.
+
+== High Level Proposal
+
+Introduce a new CRD `ManagementTask`.  A `ManagementTask` executes a management function (such as a purge) in a
+tenant namespace within EnMasse.  It would be rather similar to
+https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/[Jobs] from Kubernetes.
```

Comment:
@ctron said "could we actually implement the running of the management task as Kubernetes jobs.

Reply:
Note in document.